### PR TITLE
Revert "allow /boot on btrfs subvol or filesystem" (#1200539)

### PR DESCRIPTION
**Pull request by bcl**
This reverts commit 6dd9f042b607a6dc4e8d299009c62c5f98e6d5d4.

We really don't have support for /boot on btrfs working, turn this off
to prevent bugs.

Related: rhbz#1200539
Related: rhbz#864198
Related: rhbz#1094489

### CODE REVIEW

**Comment by vpodzime** [comment]
Looks good and wise to me.

**Comment by vpodzime** [comment]
Still looks good to me, just less wise. :)

**Comment by bcl** [comment]
Yeah, one more tweak, if /boot isn't a separate mountpoint use / instead.

File: pyanaconda/install.py [hunk 1]
```diff
@@ -107,6 +108,12 @@ def doConfiguration(storage, payload, ksdata, instClass):
     with progress_report(_("Generating initramfs")):
         payload.recreateInitrds()
 
+    # Work around rhbz#1200539, grubby doesn't handle grub2 missing initrd with /boot on btrfs
+    # So rerun writing the bootloader if this is live and /boot is on btrfs
+    if flags.flags.livecdInstall and (not ksdata.bootloader.disabled and ksdata.bootloader != "none") \
+       and isinstance(storage.mountpoints.get("/boot", storage.mountpoints.get("/")), BTRFSDevice):
```
Comment (vpodzime) [inline pyanaconda/install.py]: Would you mind moving this part ``isinstance(storage.mountpoints.get("/boot", storage.mountpoints.get("/")), BTRFSDevice)`` into a separate variable like ``boot_on_btrfs`` assigned before the if and used in the long/complex condition? I'd find that better readable.

Comment (bcl) [inline pyanaconda/install.py]: will do.

**Comment by bcl** [comment]
Pushed to master and f22-branch

**Pull request closed without merging**